main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
```

Comment:
This creates an unnecessary dependency on some global constant. Please pass the number of collectors in the constructor.

Reply:
added to my TODO's list ;)